dynamic_programming/coin_change.py

```diff
@@ -7,20 +7,29 @@
 """
 
 
-def dp_count(S, m, n):
+def dp_count(S, n):
     """
-    >>> dp_count([1, 2, 3], 3, 4)
+    >>> dp_count([1, 2, 3], 4)
     4
-    >>> dp_count([1, 2, 3], 3, 7)
+    >>> dp_count([1, 2, 3], 7)
     8
-    >>> dp_count([2, 5, 3, 6], 4, 10)
+    >>> dp_count([2, 5, 3, 6], 10)
     5
-    >>> dp_count([10], 1, 99)
+    >>> dp_count([10], 99)
     0
-    >>> dp_count([4, 5, 6], 3, 0)
+    >>> dp_count([4, 5, 6], 0)
     1
+    >>> dp_count([1, 2, 3], -5)
+    0
+    >>> dp_count([4, 5, 6], -4)
+    0
+    >>> dp_count([2, 5, 3, 6], -10)
+    0
+    >>> dp_count([4], -4)
+    0
```

Comment:
```suggestion
    >>> dp_count([1, 2, 3], -5)
    0
```
I think 1 test case for this is enough.

Reply:
@poyea You said to add some more test cases in the previous comment?

Reply:
I mean to test the `< 0` line, one test case is enough. You may add some larger test cases too to demonstrate the correctness.